content/docs/source-git/work-with-source-git/pull-upstream-fixes.md

```diff
@@ -0,0 +1,95 @@
+---
+title: "Pull fixes from the upstream"
+date: 2021-01-26T23:59:59+01:00
+draft: false
+weight: 2
+---
+
+# Pull fixes from the upstream
+
+One of the common tasks of being a downstream maintainer is to pull fixes from
+the upstream codebase. This starts by creating a new bugzilla with an issue in
+the distribution which requires a code-change in the affected package. The
+maintainer needs to pull it from upstream when:
+1. The fix is not released yet by the upstream - this implies pulling the fix
+   from the main development upstream branch.
+2. If the fix is already released, one can either update to that upstream
+   release or only pull commits with the fix. Please consult [Fedora Packaging
+   Guidelines](https://docs.fedoraproject.org/en-US/packaging-guidelines/) if
+   you are unsure how to proceed.
+
+
+## Adding changes
+
+Since your source-git repository shares git history with upstream, you can
+easily cherry-pick commits which you want to have in the downstream and packit
+is then able to turn those commits into patch files with the ability to
+configure the whole process.
+
+As an example, let's try to do this with systemd, in the example below we'd be
+using https://gitlab.com/packit-service/src/systemd.
+```
+$ git remote -v
+origin      git@gitlab.com:TomasTomecek/systemd-stable.git (push)
+source-git  git@gitlab.com:packit-service/src/systemd.git (push)
+upstream    git@github.com:systemd/systemd-stable.git (push)
+```
+
+We have 3 remotes in the repo:
+1. `origin` — a fork of the source-git repo
+2. `source-git` — the official source-git repo where the downstream maintenance happens
+3. `upstream` — the official upstream repository
+
+With this setup, we can fetch ref from the `upstream` remote, cherry-pick commits
+of our choice, push them to the fork `origin` and open a merge request against the
+repository referenced as `source-git` remote.
+
+
+### Controlling the patch process
+
+Packit recognizes that the upstream commits are meant to be downstream patches:
+1. Packit generates patch files from the commits via `git format-patch`.
+2. It also adds them into the spec file as new `PatchXYZ` entries.
+
+If that doesn't work for you, you can tailor the patch process. The main
+downside is that with the default process you cannot name the patch file nor
+control where exactly should packit place the `Patch123: 123.patch` line in the
+spec file. There is a way though how you can do all of that yourself. You can
+set patch metadata in a commit message which packit will then read and take
+into account:
+* `patch_name` — name of the patch (e.g. `patch_name: my-fancy.patch`)
+* `present_in_specfile:` — when set to `true`, it indicates the patch is set in
```

Comment:
Is this required? What is the default?

Reply:
```suggestion
* `present_in_specfile` — when set to `true`, it indicates the patch is set in
```

Reply:
the default is "false" which means that packit will try to add the patch in the specfile. That may result into a borked spec file with one Patch being defined twice (or applied twice) - we could make packit smarter about this